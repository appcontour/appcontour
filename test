#!/bin/bash
#

list=`ls tests/test.??`

tmpfile=test$$.tmp

for file in $list
do
  test=`echo $file | cut -f2 -d\.`
  echo "Running test $test..."
  source $file 2>/dev/null >$tmpfile
  exitcode=$?

  if [ -f ${file}.out ]
  then
    diff $tmpfile ${file}.out
  else
    echo "...new test, copying out file"
    cp $tmpfile ${file}.out
  fi
  rm $tmpfile
  echo "$exitcode" >$tmpfile
  if [ -f ${file}.ec ]
  then
    diff $tmpfile ${file}.ec
  else
    echo "...new test, copying ec file"
    cp $tmpfile ${file}.ec
  fi
  rm $tmpfile
done
