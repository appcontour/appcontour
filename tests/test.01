./contour applyrule n1 <examples/tre_sfere.morse
