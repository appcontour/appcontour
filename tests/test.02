#./contour applyrule n1 <examples/three_spheres.morse | ./contour applyrule n1 | ./contour applyrule n1 | ./contour applyrule n1
./contour applyrule k2 <examples/three_spheres.morse | ./contour applyrule k2 | ./contour applyrule k2 | ./contour applyrule k2
