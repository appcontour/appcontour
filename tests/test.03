./contour applyrule n1 <examples/tre_sfere.morse | ./contour applyrule n1 | ./contour applyrule n1
