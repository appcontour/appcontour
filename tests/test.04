./contour ishuffman -q examples/two_rings.morse
./contour iscontour -q examples/immersed/boysurface.morse
