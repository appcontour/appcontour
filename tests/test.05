./contour countcc <examples/tre_sfere.morse
