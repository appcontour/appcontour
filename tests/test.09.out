sketch {
Arc 1: (0);
Arc 2: (0);
Arc 3: (2);
Region 0 (f = 0): () (-a1);
Region 1 (f = 0): (-a2);
Region 2 (f = 4): (+a3);
Region 3 (f = 2): (+a1) (+a2) (-a3);
}
